File: prm393_g3_frontend/lib/features/products/domain/entities/product.dart
import 'package:equatable/equatable.dart';

class Product extends Equatable {
  const Product({
    required this.id,
    required this.name,
    required this.brandId,
    required this.categoryId,
    this.description,
    required this.status,
    this.brandName,
    this.categoryName,
  });

  final String id;
  final String name;
  final String brandId;
  final String categoryId;
  final String? description;
  final String status; // 'ACTIVE' or 'INACTIVE'
  final String? brandName;
  final String? categoryName;

  @override
  List<Object?> get props => [id, name, brandId, categoryId, description, status, brandName, categoryName];
}

File: prm393_g3_backend/src/models/product.js
import mongoose from 'mongoose';

const productSchema = new mongoose.Schema(
  {
    name: { type: String, required: true },
    // Use string codes like "brand_001", "category_001" to match existing data
    brandId: { type: String, ref: 'brand', required: true },
    categoryId: { type: String, ref: 'categories', required: true },
    description: { type: String },
    status: { type: String, enum: ['ACTIVE', 'INACTIVE'], default: 'ACTIVE' },
  },
  { timestamps: true },
);

// Use default collection name 'products' (matches MongoDB)
const Product = mongoose.model('product', productSchema);

export default Product;